src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -6,6 +6,49 @@
 from ._util import ParametersContext, patch_arg_make_required
 
 
+with ParametersContext(command='sql db') as c:
+    c.register_alias('database_name', ('--name', '-n'))
+
+with ParametersContext(command='sql db data-warehouse') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql db replication-link') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql db restore-point') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql db service-tier-advisor') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql db transparent-data-encryption') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql elastic-pools') as c:
+    c.register_alias('elastic_pool_name', ('--name', '-n'))
+
+with ParametersContext(command='sql elastic-pools db') as c:
+    c.register_alias('elastic_pool_name', ('--elastic-pool-name',))
+    c.register_alias('database_name', ('--name', '-n'))
+
+with ParametersContext(command='sql elastic-pools recommended') as c:
+    c.register_alias('recommended_elastic_pool_name', ('--name', '-n'))
+
+with ParametersContext(command='sql elastic-pools recommended db') as c:
+    c.register_alias('recommended_elastic_pool_name', ('--recommended-elastic-pool-name',))
+    c.register_alias('database_name', ('--name', '-n'))
+
+with ParametersContext(command='sql server') as c:
+    c.register_alias('server_name', ('--name', '-n'))
+
+with ParametersContext(command='sql server firewall') as c:
+    c.register_alias('server_name', ('--server-name',))
+    c.register_alias('firewall_rule_name', ('--name', '-n'))
+
+with ParametersContext(command='sql server service-objective') as c:
```

Comment:
Overall this seems like more code than would be required by simply using standard `register_cli_argument(...)` statements.